saleor/plugins/webhook/tasks.py

```diff
@@ -144,27 +150,75 @@ def send_webhook_using_google_cloud_pubsub(
     )
 
 
+def _send_webhook_to_target(parts, target_url, message, domain, signature, event_type):
+    scheme = parts.scheme.lower()
+    if scheme in [WebhookSchemes.HTTP, WebhookSchemes.HTTPS]:
+        send_webhook_using_http(target_url, message, domain, signature, event_type)
+    elif scheme == WebhookSchemes.AWS_SQS:
+        send_webhook_using_aws_sqs(target_url, message, domain, signature, event_type)
+    elif scheme == WebhookSchemes.GOOGLE_CLOUD_PUBSUB:
+        send_webhook_using_google_cloud_pubsub(
+            target_url, message, domain, signature, event_type
+        )
+    else:
+        raise ValueError("Unknown webhook scheme: %r" % (parts.scheme,))
+
+
 @app.task(
+    bind=True,
     autoretry_for=(RequestException,),
     retry_backoff=10,
     retry_kwargs={"max_retries": 5},
-    compression="zlib",
 )
-def send_webhook_request(webhook_id, target_url, secret, event_type, data):
+def send_webhook_request(
+    self, webhook_id, target_url, secret, event_type, event_payload_id
+):
+    try:
+        data = EventPayload.objects.get(id=event_payload_id)
+    except EventPayload.DoesNotExist as exc:
+        EventTask.objects.create(
+            task_id=self.request.id,
+            event_payload_id=event_payload_id,
+            status=JobStatus.FAILED,
+            error=str(exc),
+            duration=0,
+            event_type=event_type,
+            webhook_id=webhook_id,
+        )
+        task_logger.error(
```

Comment:
This function is quite long, maybe we can create something like `_get_event_payload` method for this part of the code?